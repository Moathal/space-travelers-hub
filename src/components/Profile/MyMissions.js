import React from 'react';
import { useSelector } from 'react-redux';
import MyMission from './MyMission';

function MyMissions() {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div>
      <h2>My Missions</h2>
      {joinedMissions.map((mission) => (
        <MyMission key={mission.id} name={mission.mission_name} />
      ))}
    </div>
  );
}

export default MyMissions;
