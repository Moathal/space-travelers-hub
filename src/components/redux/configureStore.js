import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions';
import rocketsReducer from './rockets';

export default configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice,
  },
});
