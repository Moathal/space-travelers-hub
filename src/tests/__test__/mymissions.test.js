import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../components/redux/configureStore';
import MyMissions from '../../components/Profile/MyMissions';

it('renders MyMissions correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <MyMissions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
