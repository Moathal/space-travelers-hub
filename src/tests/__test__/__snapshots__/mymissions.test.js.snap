// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MyMissions correctly 1`] = `
<div>
  <h2>
    My Missions
  </h2>
</div>
`;
